apis/validator/prepare_beacon_proposer.yaml

```diff
@@ -0,0 +1,45 @@
+post:
+  operationId: "prepareBeaconProposer"
+  summary: Provide beacon node with proposals for the given validators.
+  description: |
+    Prepares the beacon node for potential proposers by supplying information
+    required when proposing blocks for the given validators.  The information
+    supplied for each validator index is considered persistent until overwritten
+    by new information for the given validator index, or until the beacon node
+    restarts.
```

Comment:
I'm not sure we should require this information to be persisted until restart.  It seems reasonable for a beacon node to hold this information for a couple of epochs but expect it to be regularly updated - otherwise it winds up preparing blocks for validators that may have long since stopped using this node and the only way to stop that would be to restart the beacon node.

I'd suggest requiring it be maintained for a minimum of 2 epochs - if the VC refreshes each epoch it then handles restarts pretty well but there's "overlap" so the previous notification doesn't expire before the next one arrives.

Reply:
The concern with implicit expiry is that it isn't obvious to the validator client that it happens, or when it happens.  Is the relatively low chance that the beacon node prepares a block for a validator that no longer cares enough of a concern compared to that of the beacon node not knowing which fee recipient to provide due to a tardy validator client refresh?  I would also assume that most validators would be unlikely to change their fee recipient address so it would seem that we would be resending the same data every epoch for no real benefit.

If implicit expiry is considered desirable then we should be clear in the documentation as to exactly what that means.  I'm not opposed to the change, just want to make sure that we're happy with the resultant requirement.

Reply:
I think that's why we'd want to have a minimum time the information is persisted for - that way validators do know that they have to resend the info within that amount of time. And a tardy information refresh isn't going to be an issue as long as there's some overlap - bear in mind the VC already has to call the BN at very precise timings to create blocks and attestations.

I don't mind if it's a reasonably long period but I do feel quite uncomfortable about persisting this with no way to reset it other than a restart.  The key management API is built around the assumption that keys will be added and removed while the VC is running so it seems wrong to then design the BN API on the assumption that validators never move nodes. I agree it's not common and the impact is just extra load on the EL but it also seems pretty simple to avoid.

Reply:
I have updated the wording so that the data is transient, lasting for 2 full epochs plus whatever is left of the epoch in which the submission is made.  This gives the validator clients time to refresh the information each epoch, with the ability to miss one (or two) and still keep the beacon node up-to-date, but also decays quickly enough that it shouldn't cause lots of wasted cycles nodes generating payloads for blocks that will never be requested.